app/assets/javascripts/components/article_finder/article_finder.jsx

```diff
@@ -17,6 +18,12 @@ const ArticleFinder = createReactClass({
     };
   },
 
+  componentWillMount() {
+    if (this.props.course_id) {
```

Comment:
You can also use `state.assignments.loading` to detect whether the assignments are already loaded to avoid a duplicate fetch (as in `MyArticles` and `ArticlesHandler`). If the user navigated here from the Articles tab, then they likely will be already in the store.